kafka/src/main/java/com/opentable/logging/KafkaAppender.java

```diff
@@ -86,9 +86,7 @@ protected void append(ILoggingEvent eventObject)
             throw Throwables.propagate(e);
         }
 
-        final byte[] key = partitionShufflerKey.get();
-        key[0]++;
-        producer.send(new KeyedMessage<>(topic, key, out.toByteArray()));
+        producer.send(new KeyedMessage<>(topic, keyGenerator.next(), out.toByteArray()));
```

Comment:
I believe there is only one KafkaAppender instance running usually (a singleton in fact), so generally speaking that won't be much of an issue. I agree though, having multiple instances consuming from an internal logging queue would make the distribution of messages a little weird.